packages/search/src/util/aggregations/merge-aggregations.ts

```diff
@@ -0,0 +1,159 @@
+/**
+ * Interface for an Aggregation returned from search request,
+ * where label is a value for a fieldname type for which aggregations were requested
+ * and aggValue is the aggregation value
+ */
+interface IAggregation {
+  label: string;
+  aggValue: number;
+}
+
+/**
+ * Interface for an list of Aggregations requested for a particular fieldname
+ * Ex. { fieldName: 'access', aggregations: [ { public: 2, private: 5, ... } ] }
+ */
+export interface IAggregationResult {
+  fieldName: string;
+  aggregations: IAggregation[];
+}
+
+/**
+ * A function used when to merge the same aggregation across multiple aggregation sources.
+ * Implementations can handle summation of two aggregation values (as with COUNT and SUM),
+ * but can also handle MIN/MAX
+ */
+type mergeFunc = (aggValueOne: number, aggValueTwo: number) => number;
+
+/**
+ * Alias types helpful for readability when processing list of aggregations results
+ */
+type AggregationMap = Record<string, number>;
+type AggregationResultMap = Record<string, AggregationMap>;
+
+/**
+ * The default merge function for merging aggregations. Simply sums the aggregations
+ * @param aggValueOne the value of one aggregatiom
+ * @param aggValueTwo the value of the second aggregation
+ * @returns the sum of the two values
+ */
+const defaultMergeFunc = (aggValueOne: number, aggValueTwo: number) => {
+  return aggValueOne + aggValueTwo;
+};
+
+/**
+ * Function to merge multiple aggregations results from different result sets. Explicitly assumed
+ * that sets can contain overlapping aggregations that should be merged. Also explicitly assumed
+ * that they contain aggregation values that are undefined or null (0 is valid),
+ * and/or contain entire aggregation result sets that are empty or falsey
+ *
+ * @param aggs a list of aggregation result sets from different requests/sources
+ * @param mergeFunction a merge function used to merge aggregation values across result sets
+ * @returns a list of merged aggregations
+ */
+export function mergeAggregations(
+  aggs: IAggregationResult[][] = [],
+  mergeFunction: mergeFunc = defaultMergeFunc
+): IAggregationResult[] {
+  if (!Array.isArray(aggs) || aggs.length === 0) {
+    return [];
+  }
+
+  const mergedMap: AggregationResultMap = aggs.reduce(
+    (mergedAggs: AggregationResultMap, aggResultList: IAggregationResult[]) => {
```

Comment:
can't say i'm a huge fan of having this reduction done via side effects in the helper fn--wouldn't it be more clear/testable to pass in the accumulator, and have the reduction return an object instead of having it modify mergedAggs and returrninig that?

maybe this is just the voice of Dave speaking through me

Reply:
Yeah I definitely recognize the complexity of custom merging of objects at two different prop levels. I mulled on this back and forth when I initially implemented. Ultimately I reasoned that [reducing an array to an object of grouped key-value pairs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce#grouping_objects_by_a_property) was common enough to justify. I re-examined this morning and believe I've done a more suitable refactor.

Keep in mind that, while there are reasons we don't include it, lodash has a [function](https://lodash.com/docs/4.17.15#mergeWith) that accomplishes custom object merging